### PR TITLE
feat: add better prompt marker in REPL's multi-line mode

### LINKED ISSUE #1763: [RFC]: add prompt marker to make multi-line code more readable in the REPL
### Description

This RFC proposes adding a prompt marker to make multi-line code more readable in the REPL by separating the prompt from the code more clearly.

**Before:**

image

**After:**

after

### Related Issues

Related Issue https://github.com/stdlib-js/google-summer-of-code/issues/1

### Questions

No.

### Other

Inspired by IPython..

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by Snehil-Shah**
Resolves #1763

## Description

> What is the purpose of this pull request?

This pull request:

-   adds a better prompt when adding a new line to align with the rest of the code block

after

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1763
-   relates to https://github.com/stdlib-js/google-summer-of-code/issues/1

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/repl/lib/display_prompt.js [hunk 1]
```diff
@@ -26,6 +26,11 @@ var repeat = require( '@stdlib/string/repeat' );
 var leadingWhitespaceRegExp = require( './regexp_leading_whitespace.js' );
 
 
+// VARIABLES //
+
+var BASE_PROMPT = '...: ';
```
Comment (kgryte) [inline lib/node_modules/@stdlib/repl/lib/display_prompt.js:31]: How would this work when a user provides a custom prompt? This assumes that the prompt is of the form `In [%d]:`, but that is not necessarily the case. See https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/repl#repl-options-.

Comment (kgryte) [reply lib/node_modules/@stdlib/repl/lib/display_prompt.js:31]: E.g., the prompt could be changed to `> ` or `$ ` or something else entirely. In which case, this implementation is making an assumption about the form of the prompt which is not guaranteed to be true.

Comment (Snehil-Shah) [reply lib/node_modules/@stdlib/repl/lib/display_prompt.js:31]: You are right, I messed up there. One solution can be, the prompt can just consist of spaces/dots the length of the input prompt, the lines would still align , there would be no `...:` in the prompt

Comment (Snehil-Shah) [reply lib/node_modules/@stdlib/repl/lib/display_prompt.js:31]: @kgryte something like this:

image

Comment (kgryte) [reply lib/node_modules/@stdlib/repl/lib/display_prompt.js:31]: I think this returns us to whether a continuation prompt is necessary. For example, if we support SHIFT+ENTER or some other modifier, then

```
In [5]:
function func() {
	console.log( 'hello world' );
}
```

should be possible which achieves alignment, without requiring special prompt handling or the loss of terminal columns with unused whitespace. For the latter in particular, while some terminal applications are good in handling text overflow, others are not, thus making horizontal real estate something of a precious commodity. If we followed IPython and provided a continuation prompt which indents based on the parent prompt, we limit full use of all possible real estate.

Comment (Snehil-Shah) [reply lib/node_modules/@stdlib/repl/lib/display_prompt.js:31]: Makes sense.. I'll close this issue and PR👍

**Pull request closed without merging**